src/mscorlib/src/System/BitConverter.cs

```diff
@@ -339,21 +339,25 @@ public static bool ToBoolean(byte[] value, int startIndex)
             return value[startIndex] != 0;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)] // This should compile down to nothing or roughly 1 instruction, depending on the register holding value
```

Comment:
I don't think that this comment is right. Have you looked at the generated code?
I would expect this to compile to more than one instruction - spill the value to stack, fetch the value from stack as integer.